core/src/main/java/com/datastax/oss/driver/internal/core/metadata/NodeStateManager.java

```diff
@@ -170,75 +172,87 @@ private void onDebouncedTopologyEvent(TopologyEvent event) {
         return;
       }
       LOG.debug("[{}] Processing {}", logPrefix, event);
-      DefaultNode node = (DefaultNode) metadataManager.getMetadata().getNodes().get(event.address);
+      Optional<Node> maybeNode = metadataManager.getMetadata().findNode(event.broadcastRpcAddress);
       switch (event.type) {
         case SUGGEST_UP:
-          if (node == null) {
+          if (maybeNode.isPresent()) {
+            DefaultNode node = (DefaultNode) maybeNode.get();
+            if (node.state == NodeState.FORCED_DOWN) {
+              LOG.debug("[{}] Not setting {} UP because it is FORCED_DOWN", logPrefix, node);
+            } else if (node.distance == NodeDistance.IGNORED) {
+              setState(node, NodeState.UP, "it is IGNORED and an UP topology event was received");
+            }
+          } else {
             LOG.debug(
-                "[{}] Received UP event for unknown node {}, adding it", logPrefix, event.address);
-            metadataManager.addNode(event.address);
-          } else if (node.state == NodeState.FORCED_DOWN) {
-            LOG.debug("[{}] Not setting {} UP because it is FORCED_DOWN", logPrefix, node);
-          } else if (node.distance == NodeDistance.IGNORED) {
-            setState(node, NodeState.UP, "it is IGNORED and an UP topology event was received");
+                "[{}] Received UP event for unknown node {}, adding it",
```

Comment:
In what situation it may happen that we are receiving event but we cannot identify a specific node?

Reply:
Normally never, the only way is if the driver metadata somehow got out of sync.